impedancefitter/readin.py

```diff
@@ -23,6 +23,65 @@
 
 logger = logging.getLogger(__name__)
 
+def _readin_Data_from_formated_dataframe(formated_df, minimumFrequency=None, maximumFrequency=None):
+    # sort frequencies to be on the safe side
+    tmp = formated_df.values
+    values = tmp[tmp[:, 0].argsort()]
+
+    # filter values, so that only the ones in a certain range get taken.
+    filteredvalues = np.empty((0, values.shape[1]))
+    if minimumFrequency is None:
+        minimumFrequency = values[0, 0]
+    if maximumFrequency is None:
+        maximumFrequency = values[-1, 0]
+    logger.info("minimumFrequency is {}".format(minimumFrequency))
+    logger.info("maximumFrequency is {}".format(maximumFrequency))
+
+    for i in range(values.shape[0]):
+        if np.greater_equal(values[i, 0], minimumFrequency):
+            if np.less_equal(values[i, 0], maximumFrequency):
+                bufdict = values[i]
+                bufdict.shape = (1, bufdict.shape[0])  # change shape so it can be appended
+                filteredvalues = np.append(filteredvalues, bufdict, axis=0)
+            else:
+                break
+    values = filteredvalues
+
+    f = values[:, 0]
+    omega = 2. * np.pi * f
+    # construct complex-valued array from float data
+    zarray = np.zeros((np.int32((values.shape[1] - 1) / 2), values.shape[0]), dtype=np.complex128)
+
+    for i in range(np.int32((values.shape[1] - 1) / 2)):  # will always be an int(always real and imag part)
+        zarray[i] = values[:, (i * 2) + 1] + 1j * values[:, (i * 2) + 2]
+    return omega, zarray
+
+
+def readin_Data_from_dataframe(df, format, minimumFrequency=None, maximumFrequency=None):
```

Comment:
I think this function is enough.
Just take the dataframe, sort it by frequency, filter all frequencies that are less / greater than minimum / maximum frequency and then combine real and imaginary part to a complex-valued array.
Or am I missing something?